### DISCUSSION TITLE
What is reason for showing IWindowingPlatform is not registered error.

### DISCUSSION THREAD

**Discussion by Aswathips** [AvaloniaUI/Avalonia#8033] [Q&A] [answered]
Hello, I'am new to avalonia . Iam trying to develop an desktop application using avalonia. In my application the main window is the login form. actually I don't want to view the Login form  always . I keep the user logged in until the user click logout.
My app runs good when I start with the login form. But in other cases (means running the app without showing the login form)   it throws IWindowingPlatform is not registered.

                              Exception thrown: 'System.Exception' in Avalonia.Controls.dll
                              An unhandled exception of type 'System.Exception' occurred in Avalonia.Controls.dll
                              Could not CreateWindow(): IWindowingPlatform is not registered.

Can anyone please explain me why it throws the exception and how can I solve it.
 Please help me....

**Comment by timunie** [comment]
Hard to say without any code. I can just guess that you need to open the window differently. 

Best would be a small sample App at GitHub, so one can show you which line is causing the issue.

Happy coding
Tim

**Reply by Aswathips** [reply]
I open the window like this,

           string dir = @"E:\temp\Demo";
            AboutApp about = new AboutApp();
            if (Directory.Exists(dir))
            {
                about.Show();           //throw exception
            }
            else
            {
                BuildAvaloniaApp()
                 .StartWithClassicDesktopLifetime(args);
                
               about.Show();
            }

Sample work : https://github.com/Aswathips/DemoWork
Can you please suggest me the other ways to open the window.

**Comment by timunie** [comment, marked as answer]
Hi @Aswathips 

thanks for the sample code. I provided a fix for you: https://github.com/Aswathips/DemoWork/pull/1

your issue is, that you don't call `BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);` which needs to run before any window can show up. 

Happy coding
Tim

**Reply by Aswathips** [reply]
Thank you.
